tests/test_context.py

```diff
@@ -101,6 +116,163 @@ def test_get_trace_empty(self):
+            child._parent = root
+            child._finished = True
+            ctx.add_span(child)
+            ctx.close_span(child)
+
+        with self.override_partial_flush(ctx, enabled=True, min_spans=1):
+            trace, sampled = ctx.get()
+
+        self.assertIsNotNone(trace)
+        self.assertIsNotNone(sampled)
+
+        self.assertEqual(len(trace), 5)
+        self.assertEqual(
+            set(['child_0', 'child_1', 'child_2', 'child_3', 'child_4']),
+            set([span.name for span in trace])
+        )
+
+        # Ensure we clear/reset internal stats as expected
+        self.assertEqual(ctx._finished_spans, 0)
+        self.assertEqual(ctx._trace, [root])
+        with self.override_partial_flush(ctx, enabled=True, min_spans=5):
+            trace, sampled = ctx.get()
+            self.assertIsNone(trace)
+            self.assertIsNone(sampled)
+
+    def test_partial_flush_too_few(self):
+        """
+        When calling `Context.get`
+            When partial flushing is enabled
+            When we do not have enough finished spans to flush
+                We return no spans
+        """
+        tracer = get_dummy_tracer()
+        ctx = Context()
+
+        # Create a root span with 5 children, all of the children are finished, the root is not
+        root = Span(tracer=tracer, name='root')
+        ctx.add_span(root)
+        for i in range(5):
+            child = Span(tracer=tracer, name='child_{}'.format(i), trace_id=root.trace_id, parent_id=root.span_id)
+            child._parent = root
+            child._finished = True
+            ctx.add_span(child)
+            ctx.close_span(child)
+
+        # Test with having 1 too few spans for partial flush
+        with self.override_partial_flush(ctx, enabled=True, min_spans=6):
+            trace, sampled = ctx.get()
+
+        self.assertIsNone(trace)
+        self.assertIsNone(sampled)
+
+        self.assertEqual(len(ctx._trace), 6)
+        self.assertEqual(ctx._finished_spans, 5)
+        self.assertEqual(
+            set(['root', 'child_0', 'child_1', 'child_2', 'child_3', 'child_4']),
+            set([span.name for span in ctx._trace])
+        )
+
+    def test_partial_flush_remaining(self):
+        """
+        When calling `Context.get`
+            When partial flushing is enabled
+            When we have some unfinished spans
+                We keep the unfinished spans around
+        """
+        tracer = get_dummy_tracer()
+        ctx = Context()
+
+        # Create a root span with 5 children, all of the children are finished, the root is not
+        root = Span(tracer=tracer, name='root')
+        ctx.add_span(root)
+        for i in range(10):
+            child = Span(tracer=tracer, name='child_{}'.format(i), trace_id=root.trace_id, parent_id=root.span_id)
+            child._parent = root
+            ctx.add_span(child)
+
+            # CLose the first 5 only
```

Comment:
```suggestion
            # Close the first 5 only
```

😉 don't feel like you have to accept this one.